cmake/VASTRegisterPlugin.cmake

```diff
@@ -412,6 +412,12 @@ function (VASTRegisterPlugin)
     VASTNormalizeInstallDirs()
   endif ()
 
+  # Enable compile commands for external plugins.
+  # Must be done before the targets are created for CMake >= 3.20.
+  if (NOT "${CMAKE_PROJECT_NAME}" STREQUAL "VAST")
+    set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+  endif ()
```

Comment:
This should already be enabled via the `VASTExportCompileCommands` function. Doesn't that work for you?

Reply:
Did you read the comment? The function must be called with a target, but the option must be set before the targets are created.

Reply:
The function does not set the option, but rather the target property which is initialized from the option (which is why the option would need to be set earlier).

Reply:
The function does not set the target property. It even has a comment:
```
  # TODO: Use the EXPORT_COMPILE_COMMANDS property when using CMake >= 3.20.
```
The approach taken here works for all currently supported CMake versions.

Reply:
Not my brightest review 🙈

Should we just bump the minimum to 3.20 then? Otherwise we make it so targets defined before `VASTRegisterPlugin(...)` that link against the plugin target are not picked up by this.

Reply:
Bumping to 3.20 would not help in this case. A plugin that defines another library beforehand is a rather weird setup and I suspect it wouldn't work with the current scaffold anyways.
I'm very much in favor of just merging this working fix and moving the focus back to user problems.

Reply:
Why wouldn't it, if you set the target property on libvast, which we publicly link plugins against? That should propagate all the way up.

I care deeply about the maintainability of our CMake scaffolding because I invested a lot of time to get it to be mostly target-based, which is why I'm questioning every new piece of directory-based logic. If we didn't have that then our plugin API wouldn't be possible in the first place.

> I'm very much in favor of just merging this working fix and moving the focus back to user problems.

By that logic this PR shouldn't have existed in the first place since it's only relevant for developers using an oddly specific build setup that we do not test in CI. I think this argument is moot.

If you make such changes that are at first sight against best practices, then at least provide comments inline for why things are moved to what appears to be the wrong place so the next person touching it doesn't move it back, i.e., anticipate questions in review comments.

Feel free to resolve once such comments are in-place.

Reply:
In added a comment with directions towards a better fix that can be implemented after we assume CMake 3.20.